weld/transforms/vectorizer.rs

```diff
@@ -162,6 +162,26 @@ pub fn predicate(e: &mut Expr<Type>) {
     });
 }
 
+/// Predicate an `If` expression by checking for if(cond, scalar1, scalar2) and transforms it to select(cond, scalar1, scalar2).
+pub fn predicate_simple_expr(e: &mut Expr<Type>) {
+    e.transform_and_continue_res(&mut |ref mut e| {
+        if !(should_be_predicated(e)) {
+            return Ok((None, true));
+        }
+
+        // This pattern checks for if(cond, scalar1, scalar2).
+        if let If { ref cond, ref on_true, ref on_false } = e.kind {
+            if let Scalar(_) = on_true.ty {
+                if let Scalar(_) = on_false.ty {
```

Comment:
@sppalkia Is this sufficient? If this contains a builder, then it must contain a result as well, and then it seems that this transformation is safe. Are there other scenarios I'm missing?

Reply:
See the note I sent